pkgs/development/python-modules/mplcursors/default.nix

```diff
@@ -0,0 +1,40 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  setuptools,
+  setuptools-scm,
+  matplotlib,
+  pytestCheckHook,
+}:
+
+buildPythonPackage rec {
+  pname = "mplcursors";
+  version = "0.6";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "anntzer";
+    repo = "mplcursors";
+    rev = "v${version}";
+    hash = "sha256-L5pJqRpgPRQEsRDoP10+Pi8uzH5TQNBuGRx7hIL1x7s=";
+  };
+
+  nativeBuildInputs = [
+    setuptools
+    setuptools-scm
+  ];
+
+  propagatedBuildInputs = [ matplotlib ];
+
+  nativeCheckInputs = [ pytestCheckHook ];
+
+  pythonImportsCheck = [ "mplcursors" ];
+
+  meta = with lib; {
+    description = "Interactive data selection cursors for Matplotlib";
+    homepage = "https://github.com/anntzer/mplcursors";
+    license = licenses.zlib;
+    maintainers = with maintainers; [ gm6k ];
```

Comment:
Thanks for reviewing and for the suggestion. Do you have more information on why the suggested `meta` expression would be preferable? I'm asking because I also had the exact opposite (moving `lib` into a `with` clause) suggested in another PR, and looking at the whole package tree, `meta = with lib; {...}` seems to be very well-used.